```
$ git clone git@github.com:eisensheng/pytest-catchlog.git
$ cd pytest-catchlog/
$ git checkout develop
Already on 'develop'
Your branch is up-to-date with 'origin/develop'.
$ tox
GLOB sdist-make: /tmp/pytest-catchlog/setup.py
py26 create: /tmp/pytest-catchlog/.tox/py26
ERROR: InterpreterNotFound: python2.6
py27 inst-nodeps: /tmp/pytest-catchlog/.tox/dist/pytest-catchlog-1.2.2.zip
py27 installed: jenkins-ci-sidekick==0.0.14,jenkins-job-builder==1.4.0,py==1.4.30,pytest==2.8.3,pytest-catchlog==1.2.2,python-rrmngmnt==0.1,rrmngmnt==0.1,wheel==0.29.0
py27 runtests: PYTHONHASHSEED='1894204925'
py27 runtests: commands[0] | py.test test_pytest_catchlog.py
WARNING:test command found but not installed in testenv
  cmd: /usr/bin/py.test
  env: /tmp/pytest-catchlog/.tox/py27
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
============================= test session starts ==============================
platform linux2 -- Python 2.7.8, pytest-2.8.3, py-1.4.31, pluggy-0.3.1
rootdir: /tmp/pytest-catchlog, inifile: 
collected 15 items

test_pytest_catchlog.py .FFFFFFFFFFFFFF

=================================== FAILURES ===================================
_____________________________ test_messages_logged _____________________________

testdir = <Testdir local('/tmp/pytest-of-lbednar/pytest-9/testdir/test_messages_logged0')>

    def test_messages_logged(testdir):
        testdir.makepyfile('''
            import sys
            import logging

            def test_foo():
                sys.stdout.write('text going to stdout')
                sys.stderr.write('text going to stderr')
                logging.getLogger().info('text going to logger')
                assert False
            ''')
        result = testdir.runpytest()
        assert result.ret == 1
        result.stdout.fnmatch_lines(['*- Captured *log call -*',
>                                    '*text going to logger*'])
E       Failed: remains unmatched: '*- Captured *log call -*', see stderr

/tmp/pytest-catchlog/test_pytest_catchlog.py:44: Failed
----------------------------- Captured stdout call -----------------------------
.... continues for all evns ...
```

with this patch it goes well

```
$ git am ../0001-tox-deploy-plugin-before-running-tests.patch 
Applying: tox: deploy plugin before running tests
$ tox
GLOB sdist-make: /tmp/pytest-catchlog/setup.py
py26 create: /tmp/pytest-catchlog/.tox/py26
ERROR: InterpreterNotFound: python2.6
py27 inst-nodeps: /tmp/pytest-catchlog/.tox/dist/pytest-catchlog-1.2.2.zip
py27 installed: jenkins-ci-sidekick==0.0.14,jenkins-job-builder==1.4.0,py==1.4.30,pytest==2.8.3,pytest-catchlog==1.2.2,python-rrmngmnt==0.1,rrmngmnt==0.1,wheel==0.29.0
py27 runtests: PYTHONHASHSEED='59576136'
py27 runtests: commands[0] | pip install -e .
Obtaining file:///tmp/pytest-catchlog
Requirement already satisfied (use --upgrade to upgrade): py>=1.1.1 in ./.tox/py27/lib/python2.7/site-packages (from pytest-catchlog==1.2.2)
Collecting pytest>=2.6 (from pytest-catchlog==1.2.2)
/tmp/pytest-catchlog/.tox/py27/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:315: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#snimissingwarning.
  SNIMissingWarning
/tmp/pytest-catchlog/.tox/py27/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:120: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Using cached pytest-2.8.7-py2.py3-none-any.whl
Installing collected packages: pytest, pytest-catchlog
  Running setup.py develop for pytest-catchlog
Successfully installed pytest-2.8.7 pytest-catchlog-1.2.2
py27 runtests: commands[1] | py.test test_pytest_catchlog.py
===================================================================================== test session starts =====================================================================================
platform linux2 -- Python 2.7.8, pytest-2.8.7, py-1.4.30, pluggy-0.3.1
rootdir: /tmp/pytest-catchlog, inifile: 
plugins: catchlog-1.2.2
collected 15 items 

test_pytest_catchlog.py ...............

================================================================================== 15 passed in 0.48 seconds ==================================================================================
```

using tox

```
$ pip show tox
---
Name: tox
Version: 2.3.1
Location: /usr/lib/python2.7/site-packages
Requires: pluggy, virtualenv, py
```

--- a/tox.ini
+++ b/tox.ini
@@ -6,4 +6,4 @@ deps =
     py==1.4.30
     pytest==2.8.3
 commands =
-    py.test {posargs:test_pytest_catchlog.py}
+    {envpython} -m py.test {posargs:test_pytest_catchlog.py}

